moto/ec2/utils.py

```diff
@@ -679,3 +679,23 @@ def filter_iam_instance_profiles(iam_instance_profile_arn, iam_instance_profile_
             instance_profile = None
 
     return instance_profile
+
+
+def describe_tag_filter(filters, instances):
+    instances = list(instances)
+    for key in filters:
+        if not key.startswith('tag:'):
+            continue
+        match = re.match(r'tag:(.*)', key)
+        if match:
+            tag_key_name = match.group(1)
+        for instance in instances:
+            need_delete = True
```

Comment:
Let's say we have numbers of resource tags and we added tag filters, we want to find resources based on the tags which are supposed to be matched. We need to skip those resources whose resource tag key does not match with the filter tag but not the value.

So this line will delete any transit gateway by default. If none key match filter tags then it won't be present in the result.